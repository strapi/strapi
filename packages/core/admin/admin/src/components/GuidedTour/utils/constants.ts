const GUIDED_TOUR_REQUIRED_ACTIONS = {
  contentTypeBuilder: {
    createSchema: 'didCreateContentTypeSchema',
    addField: 'didAddFieldToSchema',
  },
  contentManager: {
    createContent: 'didCreateContent',
  },
  apiTokens: {
    createToken: 'didCreateApiToken',
    copyToken: 'didCopyApiToken',
  },
  strapiCloud: {},
} as const;

export { GUIDED_TOUR_REQUIRED_ACTIONS };
