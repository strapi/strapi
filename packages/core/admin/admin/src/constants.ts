import { PermissionMap } from './types/permissions';

import type { StrapiAppSettingLink } from './core/apis/router';

export const ADMIN_PERMISSIONS_CE = {
  contentManager: {
    main: [],
    collectionTypesConfigurations: [
      {
        action: 'plugin::content-manager.collection-types.configure-view',
        subject: null,
      },
    ],
    componentsConfigurations: [
      {
        action: 'plugin::content-manager.components.configure-layout',
        subject: null,
      },
    ],
    singleTypesConfigurations: [
      {
        action: 'plugin::content-manager.single-types.configure-view',
        subject: null,
      },
    ],
  },
  marketplace: {
    main: [{ action: 'admin::marketplace.read', subject: null }],
    read: [{ action: 'admin::marketplace.read', subject: null }],
  },
  settings: {
    roles: {
      main: [
        { action: 'admin::roles.create', subject: null },
        { action: 'admin::roles.update', subject: null },
        { action: 'admin::roles.read', subject: null },
        { action: 'admin::roles.delete', subject: null },
      ],
      create: [{ action: 'admin::roles.create', subject: null }],
      delete: [{ action: 'admin::roles.delete', subject: null }],
      read: [{ action: 'admin::roles.read', subject: null }],
      update: [{ action: 'admin::roles.update', subject: null }],
    },
    users: {
      main: [
        { action: 'admin::users.create', subject: null },
        { action: 'admin::users.read', subject: null },
        { action: 'admin::users.update', subject: null },
        { action: 'admin::users.delete', subject: null },
      ],
      create: [{ action: 'admin::users.create', subject: null }],
      delete: [{ action: 'admin::users.delete', subject: null }],
      read: [{ action: 'admin::users.read', subject: null }],
      update: [{ action: 'admin::users.update', subject: null }],
    },
    webhooks: {
      main: [
        { action: 'admin::webhooks.create', subject: null },
        { action: 'admin::webhooks.read', subject: null },
        { action: 'admin::webhooks.update', subject: null },
        { action: 'admin::webhooks.delete', subject: null },
      ],
      create: [{ action: 'admin::webhooks.create', subject: null }],
      delete: [{ action: 'admin::webhooks.delete', subject: null }],
      read: [
        { action: 'admin::webhooks.read', subject: null },
        // NOTE: We need to check with the API
        { action: 'admin::webhooks.update', subject: null },
        { action: 'admin::webhooks.delete', subject: null },
      ],
      update: [{ action: 'admin::webhooks.update', subject: null }],
    },
    'api-tokens': {
      main: [{ action: 'admin::api-tokens.access', subject: null }],
      create: [{ action: 'admin::api-tokens.create', subject: null }],
      delete: [{ action: 'admin::api-tokens.delete', subject: null }],
      read: [{ action: 'admin::api-tokens.read', subject: null }],
      update: [{ action: 'admin::api-tokens.update', subject: null }],
      regenerate: [{ action: 'admin::api-tokens.regenerate', subject: null }],
    },
    'transfer-tokens': {
      main: [{ action: 'admin::transfer.tokens.access', subject: null }],
      create: [{ action: 'admin::transfer.tokens.create', subject: null }],
      delete: [{ action: 'admin::transfer.tokens.delete', subject: null }],
      read: [{ action: 'admin::transfer.tokens.read', subject: null }],
      update: [{ action: 'admin::transfer.tokens.update', subject: null }],
      regenerate: [{ action: 'admin::transfer.tokens.regenerate', subject: null }],
    },
    'project-settings': {
      read: [{ action: 'admin::project-settings.read', subject: null }],
      update: [{ action: 'admin::project-settings.update', subject: null }],
    },
    plugins: {
      main: [{ action: 'admin::marketplace.read', subject: null }],
      read: [{ action: 'admin::marketplace.read', subject: null }],
    },
  },
} satisfies Partial<PermissionMap>;

export const HOOKS = {
  /**
   * Hook that allows to mutate the displayed headers of the list view table
   * @constant
   * @type {string}
   */
  INJECT_COLUMN_IN_TABLE: 'Admin/CM/pages/ListView/inject-column-in-table',

  /**
   * Hook that allows to mutate the CM's collection types links pre-set filters
   * @constant
   * @type {string}
   */
  MUTATE_COLLECTION_TYPES_LINKS: 'Admin/CM/pages/App/mutate-collection-types-links',

  /**
   * Hook that allows to mutate the CM's edit view layout
   * @constant
   * @type {string}
   */
  MUTATE_EDIT_VIEW_LAYOUT: 'Admin/CM/pages/EditView/mutate-edit-view-layout',

  /**
   * Hook that allows to mutate the CM's single types links pre-set filters
   * @constant
   * @type {string}
   */
  MUTATE_SINGLE_TYPES_LINKS: 'Admin/CM/pages/App/mutate-single-types-links',
};

export interface SettingsMenuLink
  extends Omit<StrapiAppSettingLink, 'Component' | 'permissions' | 'licenseOnly'> {
  licenseOnly?: boolean;
}

export type SettingsMenu = {
  admin: SettingsMenuLink[];
  global: SettingsMenuLink[];
};

export const SETTINGS_LINKS_CE = (): SettingsMenu => ({
  global: [
    {
      intlLabel: { id: 'Settings.application.title', defaultMessage: 'Overview' },
      to: '/settings/application-infos',
      id: '000-application-infos',
    },
    {
      intlLabel: { id: 'Settings.webhooks.title', defaultMessage: 'Webhooks' },
      to: '/settings/webhooks',
      id: 'webhooks',
    },
    {
      intlLabel: { id: 'Settings.apiTokens.title', defaultMessage: 'API Tokens' },
      to: '/settings/api-tokens?sort=name:ASC',
      id: 'api-tokens',
    },
    {
      intlLabel: { id: 'Settings.transferTokens.title', defaultMessage: 'Transfer Tokens' },
      to: '/settings/transfer-tokens?sort=name:ASC',
      id: 'transfer-tokens',
    },
    {
      intlLabel: {
        id: 'global.plugins',
        defaultMessage: 'Plugins',
      },
      to: '/settings/list-plugins',
      id: 'plugins',
    },
    // If the Enterprise/Cloud feature is not enabled and if the config doesn't disable it, we promote the Enterprise/Cloud feature by displaying them in the settings menu.
    // Disable this by adding "promoteEE: false" to your `./config/admin.js` file
    ...(!window.strapi.features.isEnabled(window.strapi.features.SSO) &&
    window.strapi?.flags?.promoteEE
      ? [
          {
            intlLabel: { id: 'Settings.sso.title', defaultMessage: 'Single Sign-On' },
            to: '/settings/purchase-single-sign-on',
            id: 'sso-purchase-page',
            licenseOnly: true,
          },
        ]
      : []),
    ...(!window.strapi.features.isEnabled('cms-content-history') && window.strapi?.flags?.promoteEE
      ? [
          {
            intlLabel: { id: 'Settings.content-history.title', defaultMessage: 'Content History' },
            to: '/settings/purchase-content-history',
            id: 'content-history-purchase-page',
            licenseOnly: true,
          },
        ]
      : []),
  ],

  admin: [
    {
      intlLabel: { id: 'global.roles', defaultMessage: 'Roles' },
      to: '/settings/roles',
      id: 'roles',
    },
    {
      intlLabel: { id: 'global.users', defaultMessage: 'Users' },
      // Init the search params directly
      to: '/settings/users?pageSize=10&page=1&sort=firstname',
      id: 'users',
    },
    ...(!window.strapi.features.isEnabled(window.strapi.features.AUDIT_LOGS) &&
    window.strapi?.flags?.promoteEE
      ? [
          {
            intlLabel: { id: 'global.auditLogs', defaultMessage: 'Audit Logs' },
            to: '/settings/purchase-audit-logs',
            id: 'auditLogs-purchase-page',
            licenseOnly: true,
          },
        ]
      : []),
  ],
});
