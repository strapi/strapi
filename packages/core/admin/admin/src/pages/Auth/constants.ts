import type { ComponentType } from 'react';

import { ForgotPassword } from './components/ForgotPassword';
import { ForgotPasswordSuccess } from './components/ForgotPasswordSuccess';
import { Oops } from './components/Oops';
import { Register, RegisterProps } from './components/Register';
import { ResetPassword } from './components/ResetPassword';

export type AuthType =
  | 'login'
  | 'register'
  | 'register-admin'
  | 'forgot-password'
  | 'reset-password'
  | 'forgot-password-success'
  | 'oops'
  | 'providers';

export type FormDictionary = Record<AuthType, ComponentType | ComponentType<RegisterProps>>;

export const FORMS = {
  'forgot-password': ForgotPassword,
  'forgot-password-success': ForgotPasswordSuccess,
  // the `Component` attribute is set after all forms and CE/EE components are loaded, but since we
  // are here outside of a React component we can not use the hook directly
  login: () => null,
  oops: Oops,
  register: Register,
  'register-admin': Register,
  'reset-password': ResetPassword,
  providers: () => null,
} satisfies FormDictionary;
