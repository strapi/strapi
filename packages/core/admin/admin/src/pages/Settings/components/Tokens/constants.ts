export const API_TOKEN_TYPE = 'api-token';
export const TRANSFER_TOKEN_TYPE = 'transfer-token';
