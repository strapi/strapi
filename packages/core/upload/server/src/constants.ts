const ACTIONS = {
  read: 'plugin::upload.read',
  readSettings: 'plugin::upload.settings.read',
  create: 'plugin::upload.assets.create',
  update: 'plugin::upload.assets.update',
  download: 'plugin::upload.assets.download',
  copyLink: 'plugin::upload.assets.copy-link',
  configureView: 'plugin::upload.configure-view',
};

const ALLOWED_SORT_STRINGS = [
  'createdAt:DESC',
  'createdAt:ASC',
  'name:ASC',
  'name:DESC',
  'updatedAt:DESC',
  'updatedAt:ASC',
];

const ALLOWED_WEBHOOK_EVENTS = {
  MEDIA_CREATE: 'media.create',
  MEDIA_UPDATE: 'media.update',
  MEDIA_DELETE: 'media.delete',
};

const FOLDER_MODEL_UID = 'plugin::upload.folder';
const FILE_MODEL_UID = 'plugin::upload.file';
const API_UPLOAD_FOLDER_BASE_NAME = 'API Uploads';

export {
  ACTIONS,
  FOLDER_MODEL_UID,
  FILE_MODEL_UID,
  API_UPLOAD_FOLDER_BASE_NAME,
  ALLOWED_SORT_STRINGS,
  ALLOWED_WEBHOOK_EVENTS,
};
