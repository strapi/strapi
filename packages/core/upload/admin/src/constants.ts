import { getTrad } from './utils';

export enum AssetType {
  Video = 'video',
  Image = 'image',
  Document = 'doc',
  Audio = 'audio',
}

export enum AssetSource {
  Url = 'url',
  Computer = 'computer',
}

export const PERMISSIONS = {
  // This permission regards the main component (App) and is used to tell
  // If the plugin link should be displayed in the menu
  // And also if the plugin is accessible. This use case is found when a user types the url of the
  // plugin directly in the browser
  main: [
    { action: 'plugin::upload.read', subject: null },
    {
      action: 'plugin::upload.assets.create',
      subject: null,
    },
    {
      action: 'plugin::upload.assets.update',
      subject: null,
    },
  ],
  copyLink: [
    {
      action: 'plugin::upload.assets.copy-link',
      subject: null,
    },
  ],
  create: [
    {
      action: 'plugin::upload.assets.create',
      subject: null,
    },
  ],
  download: [
    {
      action: 'plugin::upload.assets.download',
      subject: null,
    },
  ],
  read: [{ action: 'plugin::upload.read', subject: null }],
  configureView: [{ action: 'plugin::upload.configure-view', subject: null }],
  settings: [{ action: 'plugin::upload.settings.read', subject: null }],
  update: [{ action: 'plugin::upload.assets.update', subject: null, fields: null }],
};

export const tableHeaders = [
  {
    name: 'preview',
    key: 'preview',
    metadatas: {
      label: { id: getTrad('list.table.header.preview'), defaultMessage: 'preview' },
      isSortable: false,
    },
    type: 'image',
  },
  {
    name: 'name',
    key: 'name',
    metadatas: {
      label: { id: getTrad('list.table.header.name'), defaultMessage: 'name' },
      isSortable: true,
    },
    type: 'text',
  },
  {
    name: 'ext',
    key: 'extension',
    metadatas: {
      label: { id: getTrad('list.table.header.ext'), defaultMessage: 'extension' },
      isSortable: false,
    },
    type: 'ext',
  },
  {
    name: 'size',
    key: 'size',
    metadatas: {
      label: { id: getTrad('list.table.header.size'), defaultMessage: 'size' },
      isSortable: false,
    },
    type: 'size',
  },
  {
    name: 'createdAt',
    key: 'createdAt',
    metadatas: {
      label: { id: getTrad('list.table.header.createdAt'), defaultMessage: 'created' },
      isSortable: true,
    },
    type: 'date',
  },
  {
    name: 'updatedAt',
    key: 'updatedAt',
    metadatas: {
      label: { id: getTrad('list.table.header.updatedAt'), defaultMessage: 'last update' },
      isSortable: true,
    },
    type: 'date',
  },
];

export const sortOptions = [
  { key: 'sort.created_at_desc', value: 'createdAt:DESC' },
  { key: 'sort.created_at_asc', value: 'createdAt:ASC' },
  { key: 'sort.name_asc', value: 'name:ASC' },
  { key: 'sort.name_desc', value: 'name:DESC' },
  { key: 'sort.updated_at_desc', value: 'updatedAt:DESC' },
  { key: 'sort.updated_at_asc', value: 'updatedAt:ASC' },
];

export const pageSizes = [10, 20, 50, 100];

export const localStorageKeys = {
  modalView: `STRAPI_UPLOAD_MODAL_VIEW`,
  view: `STRAPI_UPLOAD_LIBRARY_VIEW`,
};

export const viewOptions = {
  GRID: 0,
  LIST: 1,
};
