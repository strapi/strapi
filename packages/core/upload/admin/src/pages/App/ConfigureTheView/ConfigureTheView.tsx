// TODO: find a better naming convention for the file that was an index file before
import * as React from 'react';

import {
  ConfirmDialog,
  useTracking,
  useNotification,
  Page,
  Layouts,
} from '@strapi/admin/strapi-admin';
import { Button, Dialog, Link } from '@strapi/design-system';
import { ArrowLeft, Check } from '@strapi/icons';
import isEqual from 'lodash/isEqual';
import { useIntl } from 'react-intl';
import { NavLink } from 'react-router-dom';

import { useConfig } from '../../../hooks/useConfig';
import { pluginId } from '../../../pluginId';
import { getTrad } from '../../../utils';

import { Settings } from './components/Settings';
import { onChange, setLoaded } from './state/actions';
import { init, initialState } from './state/init';
import { reducer } from './state/reducer';

import type { InitialState } from './state/init';
import type { Action } from './state/reducer';
import type { Configuration } from '../../../../../shared/contracts/configuration';

interface ConfigureTheViewProps {
  config: Configuration;
}

export const ConfigureTheView = ({ config }: ConfigureTheViewProps) => {
  const { trackUsage } = useTracking();
  const { formatMessage } = useIntl();
  const { toggleNotification } = useNotification();
  const { mutateConfig } = useConfig();
  const { isLoading: isSubmittingForm } = mutateConfig;

  const [showWarningSubmit, setWarningSubmit] = React.useState(false);
  const toggleWarningSubmit = () => setWarningSubmit((prevState) => !prevState);

  const [reducerState, dispatch] = React.useReducer(
    reducer,
    initialState,
    (): InitialState => init(config)
  );
  const typedDispatch: React.Dispatch<Action> = dispatch;
  const { initialData, modifiedData } = reducerState;

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    toggleWarningSubmit();
  };

  const handleConfirm = async () => {
    trackUsage('willEditMediaLibraryConfig');
    await mutateConfig.mutateAsync(modifiedData as Configuration);
    setWarningSubmit(false);
    typedDispatch(setLoaded());
    toggleNotification({
      type: 'success',
      message: formatMessage({
        id: 'notification.form.success.fields',
        defaultMessage: 'Changes saved',
      }),
    });
  };

  const handleChange = ({
    target: { name, value },
  }: {
    target: { name: keyof Configuration; value: string | number };
  }) => {
    typedDispatch(onChange({ name, value }));
  };

  return (
    <Layouts.Root>
      <Page.Main aria-busy={isSubmittingForm}>
        <form onSubmit={handleSubmit}>
          <Layouts.Header
            navigationAction={
              <Link
                tag={NavLink}
                startIcon={<ArrowLeft />}
                to={`/plugins/${pluginId}`}
                id="go-back"
              >
                {formatMessage({ id: getTrad('config.back'), defaultMessage: 'Back' })}
              </Link>
            }
            primaryAction={
              <Button
                size="S"
                startIcon={<Check />}
                disabled={isEqual(modifiedData, initialData)}
                type="submit"
              >
                {formatMessage({ id: 'global.save', defaultMessage: 'Save' })}
              </Button>
            }
            subtitle={formatMessage({
              id: getTrad('config.subtitle'),
              defaultMessage: 'Define the view settings of the media library.',
            })}
            title={formatMessage({
              id: getTrad('config.title'),
              defaultMessage: 'Configure the view - Media Library',
            })}
          />
          <Layouts.Content>
            <Settings
              data-testid="settings"
              pageSize={modifiedData.pageSize || ''}
              sort={modifiedData.sort || ''}
              onChange={handleChange}
            />
          </Layouts.Content>
          <Dialog.Root open={showWarningSubmit} onOpenChange={toggleWarningSubmit}>
            <ConfirmDialog onConfirm={handleConfirm} variant="default">
              {formatMessage({
                id: getTrad('config.popUpWarning.warning.updateAllSettings'),
                defaultMessage: 'This will modify all your settings',
              })}
            </ConfirmDialog>
          </Dialog.Root>
        </form>
      </Page.Main>
    </Layouts.Root>
  );
};
