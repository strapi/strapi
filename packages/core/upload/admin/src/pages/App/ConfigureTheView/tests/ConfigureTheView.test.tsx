import { DesignSystemProvider } from '@strapi/design-system';
import { fireEvent, render as renderRTL, waitFor } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom';

import { pageSizes, sortOptions } from '../../../../constants';
import { ConfigureTheView } from '../ConfigureTheView';

import type { Configuration } from '../../../../../../shared/contracts/configuration';

const mutateAsync = jest.fn();
jest.mock('../../../../hooks/useConfig', () => ({
  useConfig: jest.fn(() => ({
    mutateConfig: {
      mutateAsync,
    },
  })),
}));

const render = (
  config: Configuration = {
    pageSize: pageSizes[0],
    sort: sortOptions[0].value as Configuration['sort'],
  }
) => ({
  user: userEvent.setup(),
  ...renderRTL(<ConfigureTheView config={config} />, {
    wrapper: ({ children }) => (
      <IntlProvider locale="en" messages={{}}>
        <DesignSystemProvider>
          <MemoryRouter>{children}</MemoryRouter>
        </DesignSystemProvider>
      </IntlProvider>
    ),
  }),
});

describe('Upload - Configure', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('initial render', () => {
    it('renders and matches the snapshot', () => {
      const { container, getByText } = render();

      expect(getByText('Configure the view - Media Library')).toBeInTheDocument();

      expect(container).toMatchSnapshot();
    });
  });

  describe('save', () => {
    it('save renders and is initially disabled', () => {
      const { getByText, getByRole } = render();

      expect(getByText('Configure the view - Media Library')).toBeInTheDocument();
      expect(
        getByRole('button', {
          name: 'Save',
        })
      ).toBeDisabled();
    });
  });

  describe('user actions', () => {
    const testPageSize = pageSizes[1];

    it('modify settings', async () => {
      const { user, getByRole, getByText } = render();

      expect(getByRole('combobox', { name: 'Entries per page' })).toHaveTextContent('10');

      await user.click(getByRole('combobox', { name: 'Entries per page' }));
      await user.click(getByRole('option', { name: testPageSize.toString() }));

      expect(getByRole('combobox', { name: 'Entries per page' })).toHaveTextContent('20');

      expect(
        getByRole('button', {
          name: 'Save',
        })
      ).toBeEnabled();

      /**
       * using `userEvent.click` does not fire the submit event for the form :(
       * see – https://github.com/testing-library/user-event/issues/1075
       * see – https://github.com/testing-library/user-event/issues/1002
       */
      fireEvent.click(
        getByRole('button', {
          name: 'Save',
        })
      );

      await waitFor(() => {
        expect(getByText('This will modify all your settings')).toBeInTheDocument();
      });

      await user.click(getByText('Confirm'));

      expect(mutateAsync).toHaveBeenCalledTimes(1);
      expect(mutateAsync).toHaveBeenCalledWith({
        pageSize: testPageSize,
        sort: 'createdAt:DESC',
      });
    });
  });
});
