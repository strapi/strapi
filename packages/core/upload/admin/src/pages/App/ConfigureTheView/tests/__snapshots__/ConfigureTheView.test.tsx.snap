// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload - Configure initial render renders and matches the snapshot 1`] = `
.c1 {
  padding-block-end: 56px;
}

.c4 {
  padding-block-start: 24px;
  padding-block-end: 40px;
  padding-inline-start: 56px;
  padding-inline-end: 56px;
  background: #f6f6f9;
}

.c5 {
  padding-block-end: 8px;
}

.c6 {
  cursor: pointer;
}

.c11 {
  min-width: 0;
}

.c14 {
  padding-inline-start: 16px;
  padding-inline-end: 16px;
  border-radius: 4px;
  cursor: pointer;
}

.c19 {
  padding-inline-start: 56px;
  padding-inline-end: 56px;
}

.c20 {
  padding-block-start: 24px;
  padding-block-end: 24px;
  padding-inline-start: 32px;
  padding-inline-end: 32px;
  background: #ffffff;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33, 33, 52, 0.1);
}

.c27 {
  background: #ffffff;
  border-radius: 4px;
  cursor: default;
  position: relative;
  overflow: hidden;
}

.c30 {
  flex: 1;
}

.c10 {
  align-items: center;
  justify-content: space-between;
  flex-direction: row;
  display: flex;
}

.c12 {
  align-items: center;
  flex-direction: row;
  display: flex;
}

.c15 {
  gap: 8px;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: inline-flex;
}

.c22 {
  align-items: stretch;
  flex-direction: column;
  display: flex;
}

.c24 {
  gap: 4px;
  align-items: stretch;
  flex-direction: column;
  display: flex;
}

.c28 {
  gap: 16px;
  align-items: center;
  justify-content: space-between;
  flex-direction: row;
  display: flex;
}

.c31 {
  gap: 12px;
  align-items: center;
  flex-direction: row;
  display: flex;
}

.c9 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: #4945ff;
}

.c13 {
  font-weight: 600;
  font-size: 3.2rem;
  line-height: 1.25;
  color: currentcolor;
}

.c17 {
  font-size: 1.2rem;
  line-height: 1.33;
  color: currentcolor;
  font-weight: 600;
}

.c18 {
  font-size: 1.6rem;
  line-height: 1.5;
  color: #666687;
}

.c25 {
  font-size: 1.2rem;
  line-height: 1.33;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
  font-weight: 600;
}

.c32 {
  font-size: 1.4rem;
  line-height: 1.43;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
}

.c36 {
  font-size: 1.2rem;
  line-height: 1.33;
  color: #666687;
}

.c37 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c7 {
  text-decoration: none;
}

.c7:visited {
  color: inherit;
}

.c16 {
  height: 3.2rem;
  text-decoration: none;
  border: 1px solid #4945ff;
  background: #4945ff;
  color: #ffffff;
}

.c16:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c16:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c16[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
  cursor: default;
}

.c8 {
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  gap: 8px;
  position: relative;
  outline: none;
}

.c8 svg {
  font-size: 1rem;
}

.c8 svg path {
  fill: #4945ff;
}

.c8:hover>span {
  color: #7b79ff;
}

.c8:hover svg path {
  fill: #7b79ff;
}

.c8:active {
  color: #271fe0;
}

.c8:after {
  transition-property: all;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c8:focus-visible {
  outline: none;
}

.c8:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c26 {
  display: flex;
}

.c29 {
  border: 1px solid #dcdce4;
  padding-block: 8px;
  padding-inline-start: 16px;
  padding-inline-end: 12px;
  cursor: pointer;
  outline: none;
  box-shadow: none;
  transition-property: border-color,box-shadow,fill;
  transition-duration: 0.2s;
}

.c29[aria-disabled='true'] {
  color: #8e8ea9;
}

.c29:focus-visible {
  outline: none;
}

.c29:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c35 {
  display: flex;
}

.c35>svg {
  fill: #8e8ea9;
}

.c33 {
  flex: 1;
  font-size: 1.4rem;
  line-height: 2.2rem;
}

.c34 {
  display: flex;
  gap: 4px;
  flex-wrap: wrap;
}

.c3:focus-visible {
  outline: none;
}

.c21 {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  gap: 16px;
}

.c23 {
  grid-column: span 12;
  max-width: 100%;
}

.c0 {
  display: grid;
  grid-template-columns: 1fr;
}

.c2 {
  overflow-x: hidden;
}

@media (prefers-reduced-motion: no-preference) {
  .c16 {
    transition: background-color 120ms cubic-bezier(0.25, 0.46, 0.45, 0.94),color 120ms cubic-bezier(0.25, 0.46, 0.45, 0.94),border-color 200ms cubic-bezier(0.25, 0.46, 0.45, 0.94);
  }
}

@media (prefers-reduced-motion: no-preference) {

}

@media (min-width: 520px) {
  .c23 {
    grid-column: span 12;
  }
}

@media (min-width: 768px) {
  .c23 {
    grid-column: span 12;
  }
}

@media (min-width: 1080px) {
  .c23 {
    grid-column: span 6;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <main
        aria-labelledby="main-content-title"
        class="c3"
        id="main-content"
        tabindex="-1"
      >
        <form>
          <div
            style="height: 0px;"
          >
            <div
              class="c4"
              data-strapi-header="true"
            >
              <div
                class="c5"
              >
                <a
                  aria-disabled="false"
                  class="c6 c7 c8"
                  href="/plugins/upload"
                  id="go-back"
                >
                  <svg
                    fill="currentColor"
                    height="16"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M28.5 16a1.5 1.5 0 0 1-1.5 1.5H8.625l6.44 6.439a1.502 1.502 0 1 1-2.125 2.125l-9-9a1.5 1.5 0 0 1 0-2.125l9-9a1.503 1.503 0 0 1 2.125 2.125L8.625 14.5H27a1.5 1.5 0 0 1 1.5 1.5"
                    />
                  </svg>
                  <span
                    class="c9"
                  >
                    Back
                  </span>
                </a>
              </div>
              <div
                class="c10"
              >
                <div
                  class="c11 c12"
                >
                  <h1
                    class="c13"
                  >
                    Configure the view - Media Library
                  </h1>
                </div>
                <button
                  aria-disabled="true"
                  class="c14 c15 c16"
                  disabled=""
                  type="submit"
                >
                  <span
                    aria-hidden="true"
                    class="c12"
                  >
                    <svg
                      fill="currentColor"
                      height="16"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m29.061 10.061-16 16a1.5 1.5 0 0 1-2.125 0l-7-7a1.504 1.504 0 0 1 2.125-2.125L12 22.875 26.939 7.939a1.502 1.502 0 1 1 2.125 2.125z"
                      />
                    </svg>
                  </span>
                  <span
                    class="c17"
                  >
                    Save
                  </span>
                </button>
              </div>
              <p
                class="c18"
              >
                Define the view settings of the media library.
              </p>
            </div>
          </div>
          <div
            class="c19"
          >
            <div
              class="c20"
            >
              <div
                class="c21"
              >
                <div
                  class="c22 c23"
                >
                  <div
                    class="c24"
                  >
                    <label
                      class="c25 c26"
                      for=":r0:"
                      id=":r0:-label"
                    >
                      Entries per page
                    </label>
                    <div
                      aria-autocomplete="none"
                      aria-controls="radix-:r2:"
                      aria-describedby=":r0:-hint"
                      aria-expanded="false"
                      aria-labelledby=":r0:-label"
                      aria-required="false"
                      class="c27 c28 c29"
                      data-state="closed"
                      dir="ltr"
                      name="pageSize"
                      role="combobox"
                      tabindex="0"
                    >
                      <span
                        class="c30 c31"
                      >
                        <span
                          class="c32 c33"
                        >
                          <span
                            class="c34"
                          >
                            10
                          </span>
                        </span>
                      </span>
                      <span
                        class="c31"
                      >
                        <span
                          aria-hidden="true"
                          class="c35"
                        >
                          <svg
                            fill="currentColor"
                            height="16"
                            viewBox="0 0 32 32"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                    <p
                      class="c36"
                      id=":r0:-hint"
                    >
                      Number of assets displayed by default in the Media Library
                    </p>
                  </div>
                </div>
                <div
                  class="c22 c23"
                >
                  <div
                    class="c24"
                  >
                    <label
                      class="c25 c26"
                      for=":r3:"
                      id=":r3:-label"
                    >
                      Default sort order
                    </label>
                    <div
                      aria-autocomplete="none"
                      aria-controls="radix-:r5:"
                      aria-describedby=":r3:-hint"
                      aria-expanded="false"
                      aria-labelledby=":r3:-label"
                      aria-required="false"
                      class="c27 c28 c29"
                      data-state="closed"
                      dir="ltr"
                      name="sort"
                      role="combobox"
                      tabindex="0"
                    >
                      <span
                        class="c30 c31"
                      >
                        <span
                          class="c32 c33"
                        >
                          <span
                            class="c34"
                          >
                            createdAt:DESC
                          </span>
                        </span>
                      </span>
                      <span
                        class="c31"
                      >
                        <span
                          aria-hidden="true"
                          class="c35"
                        >
                          <svg
                            fill="currentColor"
                            height="16"
                            viewBox="0 0 32 32"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                    <p
                      class="c36"
                      id=":r3:-hint"
                    >
                      Note: You can override this value in the media library.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          x
        </form>
      </main>
    </div>
  </div>
  <span
    class="c37"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;
